src/components/TimelineMarker/index.js

```diff
@@ -10,18 +13,97 @@ import StyledLabel from './styled/label';
 import StyledBody from './styled/body';
+import ExpandCollapseButton from './expandCollapseButton';
+import removeItemFromArray from '../AccordionSection/removeItemFromArray';
+import isInArray from '../AccordionSection/isInArray';
 
 /**
- * The TimelineMarker displays one event of an item timeline.
+ * The TimelineMarker displays one event of an item's timeline. It's generally used to compose the ActivityTimeline component.
  * @category Layout
  */
 export default function TimelineMarker(props) {
-    const { icon, label, description, datetime, children, className, style } = props;
+    const {
+        icon,
+        name,
+        isLoading,
+        label,
+        description,
+        datetime,
+        children,
+        className,
+        style,
+    } = props;
+    const {
+        isVariantAccordion,
+        activeNames,
+        multiple,
+        privateRegisterMarker,
+        privateUnregisterMarker,
+        privateOnToggleMarker,
+    } = useContext(ActivityTimelineContext);
+    const generatedName = useUniqueIdentifier('timeline-marker');
+    const currentName = useMemo(() => name || generatedName, [generatedName, name]);
+    const stepRef = useRef();
+
+    useEffect(() => {
+        privateRegisterMarker(stepRef.current, { currentName });
+
+        return () => {
+            privateUnregisterMarker(stepRef, currentName);
+        };
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
+
+    const resolveActiveNamesWhenMultiple = useCallback(() => {
+        if (activeNames === undefined) {
+            return [currentName];
+        }
+        if (isInArray(activeNames, currentName)) {
+            return removeItemFromArray(activeNames, currentName);
+        }
+        return activeNames.concat([currentName]);
+    }, [currentName, activeNames]);
+
+    const resolveActiveNames = useCallback(() => {
+        if (multiple) {
+            return resolveActiveNamesWhenMultiple();
+        }
+        if (currentName === activeNames) {
+            return '';
+        }
+        return currentName;
+    }, [currentName, multiple, activeNames, resolveActiveNamesWhenMultiple]);
+
+    const handleToggleMarker = useCallback(
+        event => privateOnToggleMarker(event, resolveActiveNames()),
+        [privateOnToggleMarker, resolveActiveNames],
+    );
+
+    const isExpanded = useMemo(() => {
+        if (isVariantAccordion) {
+            if (multiple && Array.isArray(activeNames)) {
+                return isInArray(activeNames, currentName);
+            }
+            return activeNames === currentName;
+        }
+        return true;
+    }, [isVariantAccordion, currentName, activeNames, multiple]);
+
+    const shouldRenderChildren = useMemo(() => isExpanded && !isLoading, [isLoading, isExpanded]);
 
     return (
-        <StyledLi className={className} style={style}>
+        <StyledLi data-id="timeline-marker-li" className={className} style={style}>
```

Comment:
same ^^ here we should only execute all this new code if it is accordion variant